### PR TITLE
[IOTDB-4713] Log TException in ConfigNodeClient

### PR DESCRIPTION
**Pull request by MrQuansy**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/client/ConfigNodeClient.java [hunk 1]
```diff
@@ -441,9 +486,13 @@ public TStorageGroupSchemaResp getMatchedStorageGroupSchemas(List<String> storag
...
         }
       } catch (TException e) {
+        logger.warn(
+            "Failed to connect to ConfigNode {} from DataNode {} when executing {}",
+            configNode,
+            config.getAddressAndPort(),
+            Thread.currentThread().getStackTrace()[1].getMethodName());
         configLeader = null;
       }
-      reconnect();
```
Comment (CRZbulabula) [inline server/src/main/java/org/apache/iotdb/db/client/ConfigNodeClient.java:446]: Why getMatchedStorageGroupSchemas is different from other interfaces in this line?

**Review by CRZbulabula — COMMENTED** [review]
PTAL~

**Review by CRZbulabula — APPROVED** [review]
LGTM

**Review by wangchao316 — APPROVED** [review]
LGTM

**Pull request merged**